File: typescript/base-hmi-types/src/screens/widgets/HmiSignatureButtonSettings.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";

export enum HmiSignatureButtonOperation {
  SetNumericValue = "SetNumericValue",
  SetStringValue = "SetStringValue",
  SetBooleanValue = "SetBooleanValue",
  SetDateTimeValue = "SetDateTimeValue",
  SendCommand = "SendCommand",
  DownloadInputValues = "DownloadInputValues",
}

export class HmiSignatureButtonSettings {
  operation?: HmiSignatureButtonOperation;
  windowTitle?: HmiMultilingualText;
  operationDescription?: HmiMultilingualText;
  numericMinimum?: number;
  numericMaximum?: number;
  decimalPlaces?: number;
  command?: string;
  valueExpression?: string;
  datePartExpression?: string;
  timePartExpression?: string;
  dateTimeStringExpression?: string;
  performerAuthenticationEnabled?: boolean;
  performerGroup?: string;
  approverAuthenticationEnabled?: boolean;
  approverGroup?: string;
  defaultDomain?: string;
}

File: src/BaseHmiTypes/Screens/Widgets/HmiSignatureButtonSettings.cs
using BaseHmiTypes.Common;

namespace BaseHmiTypes.Screens.Widgets;

public enum HmiSignatureButtonOperation
{
    SetNumericValue,
    SetStringValue,
    SetBooleanValue,
    SetDateTimeValue,
    SendCommand,
    DownloadInputValues
}

public sealed class HmiSignatureButtonSettings
{
    public HmiSignatureButtonOperation? Operation { get; set; }

    public HmiMultilingualText? WindowTitle { get; set; }

    public HmiMultilingualText? OperationDescription { get; set; }

    public double? NumericMinimum { get; set; }

    public double? NumericMaximum { get; set; }

    public int? DecimalPlaces { get; set; }

    public string? Command { get; set; }

    public string? ValueExpression { get; set; }

    public string? DatePartExpression { get; set; }

    public string? TimePartExpression { get; set; }

    public string? DateTimeStringExpression { get; set; }

    public bool? PerformerAuthenticationEnabled { get; set; }

    public string? PerformerGroup { get; set; }

    public bool? ApproverAuthenticationEnabled { get; set; }

    public string? ApproverGroup { get; set; }

    public string? DefaultDomain { get; set; }
}
